File: Python/백준/Silver/1652. 누울 자리를 찾아라/누울 자리를 찾아라.py
a=int(input()); cnt1 = 0; cnt2 = 0; bb=[]
for i in range(a):
    bb.append(input())
    ck=0
    for j in range(a-1):
        if(bb[i][j]==bb[i][j+1] and bb[i][j]=='.' and ck==0):
            cnt1+=1
            ck=1
        elif(bb[i][j]=='X'):
            ck=0
for i in range(a):
    ck=0
    for j in range(a-1):
        if(bb[j][i]==bb[j+1][i] and bb[j][i]=='.' and ck==0):
            cnt2+=1
            ck=1
        elif(bb[j][i]=='X'):
            ck=0
print(cnt1,cnt2)

File: Java/백준/Silver/1652. 누울 자리를 찾아라/누울 자리를 찾아라.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int ck=0;
        int cnt1=0;
        int cnt2=0;
        String[] b = new String[a];
        for (int i=0; i<a; i++){
            b[i] = in.next();
            ck=0;
            for(int j =0; j<a-1; j++){
                if(b[i].charAt(j)==b[i].charAt(j+1) && b[i].charAt(j) == '.' && ck==0){
                    cnt1++;
                    ck++;
                }
                else if(b[i].charAt(j) == 'X'){
                    ck=0;
                }
            }
        }
        for(int i=0; i<a; i++){
            ck=0;
            for (int j=0; j<a-1; j++){
                if(b[j].charAt(i) == b[j+1].charAt(i) && b[j].charAt(i) == '.' && ck==0){
                    cnt2++;
                    ck++;
                }
                else if(b[j].charAt(i) == 'X'){
                    ck=0;
                }
            }
        }
        System.out.printf("%d %d",cnt1,cnt2);
    }
}